platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/CollectionConfig.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.solr.factory.impl;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Properties;
+import org.apache.commons.lang.math.NumberUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CollectionConfig {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CollectionConfig.class);
+
+  private final int shardCount;
+
+  private final int replicationFactor;
+
+  private final int maximumShardsPerNode;
+
+  public CollectionConfig(String collection) {
+    this(collection, Paths.get(System.getProperty("ddf.home", ""), "etc/solr/configsets"));
+  }
+
+  public CollectionConfig(String collection, Path collectionConfigs) {
+    File matchingConfig =
+        collectionConfigs.resolve(Paths.get(collection, "collection.properties")).toFile();
+    Properties collectionProps = new Properties();
+    if (matchingConfig.exists()) {
+      try {
+        collectionProps.load(new FileInputStream(matchingConfig));
```

Comment:
❓ Should we be creating this `FileInputStream` with a try with resources? `Properties::load` states 
>  The specified stream remains open after this method returns.

Reply:
Oops. Yes, I forgot to clean this up. Fixed.